src/bitcoin-cli.cpp

```diff
@@ -1078,6 +1079,68 @@ static UniValue GetNewAddress()
     return ConnectAndCallRPC(&rh, "getnewaddress", /* args=*/{}, wallet_name);
 }
 
+/**
+ * IsExclusivelyCliCommand checks if a string is a cli-command that needs to run exclusively
+ * @param[in] cliCommand  Reference to const string cli-command.
+ * @returns true if cliCommand it's included in the set of the cli-commands.
+ */
+bool IsExclusivelyCliCommand(const std::string cliCommand)
+{
+    // Initialize set with cli-commands that run exclusively
+    std::set<std::string> cliCommands{"-generate", "-netinfo", "-addrinfo", "-getinfo"};
+
+    // return if element was found or not
+    return (cliCommands.find(cliCommand) != cliCommands.end());
+}
+
+/**
+ * ValidateCliCommand checks for:
+ *  1. duplicate command line arguments
+ *  2. no multiple bitcoin-cli side commands
+ *  3. unrecognised cli-command or any other argument starting with a slash
+ * and throws an exception if any of the conditions are met.
+ * @throws std::runtime_error if there are invalid args.
+ */
+static void ValidateCliCommand(int argc, char *argv[])
+{
+    std::set<std::string> commands;
+    std::string exclusivelyCliCommand;
+    for (int i = 1; i < argc; ++i) {
+        if (IsSwitchChar(argv[i][0])) {
+            std::string command(argv[i]);
+            // Check if the command line argument has an =, in that case, ignore what's after;
+            // e.g.: -color=never; we just want to check "-color"
+            std::size_t pos = command.find("=");
+            if (pos != std::string::npos) {
+                command = command.substr(0, pos);
+            }
+
+            // Check if this command has already been seen
+            if (commands.count(command) > 0) {
+                throw std::runtime_error(strprintf("Error: duplicate command line argument %s", command));
+            }
+
+            // Check if it's a cli-command that runs exclusively e.g. a user could run:
+            // "bitcoin-cli -regtest -generate -netinfo -addrinfo -getinfo" but it's confusing,
+            // only the last one will be executed so we make sure the user specify only one
+            if (IsExclusivelyCliCommand(command)) {
+                if (exclusivelyCliCommand != "") {
+                    throw std::runtime_error(strprintf("Error: you can only run one cli-command at a time, either %s or %s", exclusivelyCliCommand, command));
+                }
+                exclusivelyCliCommand = command;
+            }
+
+            // Check if the "-" command was recognised by the ArgsManager or interpreted as a parameter
+            // e.g.: bitcoin-cli -generate 1 -rpcwallet=xxx, is an invalid call
+            if (!gArgs.IsArgSet(command) && exclusivelyCliCommand != "") {
+                throw std::runtime_error(strprintf("Error: invalid cli-command argument. If \"%s\" is a valid option try passing it before the \"%s\" command.", argv[i], exclusivelyCliCommand));
```

Comment:
Is there a reason we shouldn't just support this? Normally it is okay to pass switch options in any order.

Reply:
Perhaps we should, this issue has been here for a while it seems (#1044). I'll ping you with an alternative I've mentioned in a [comment](https://github.com/bitcoin/bitcoin/pull/26990#issuecomment-1411046166) above (3rd paragraph).